triehash/src/lib.rs

```diff
@@ -240,19 +232,18 @@ where
 	stream.begin_list(17);
 
 	// if first key len is equal to prefix_len, move to next element
-	let mut begin = match pre_len == key.len() {
-		true => 1,
-		false => 0
-	};
+	let mut begin = if pre_len == key.len() { 1 } else { 0 };
 
 	// iterate over all possible nibbles
 	for i in 0..16 {
 		// count how many successive elements have same next nibble
-		let len = match begin < input.len() {
-			true => input[begin..].iter()
-				.take_while(| pair | pair.0.as_ref()[pre_len] == i )
-				.count(),
-			false => 0
+		let len = if begin < input.len() {
+			input[begin..]
```

Comment:
I think the `if-else` can be removed by using the following instead:

```rustc
input
   .iter()
   .skip(begin)
   .take_while(|pair| pair.0.as_ref()[pre_len] == i)
   .count()
```

Reply:
Yeah, that looks good.